### ISSUE TITLE
interactively pipe stream to cz check

### ISSUE THREAD

**Issue by guihao-liang** [commitizen-tools/commitizen#200]
## Description
similar to [commitlint](https://commitlint.js.org/#/), I want to interactively try commitizen with pipe.

```bash
echo "fix(parser): appending extra parse arms" | cz check
```
## Possible Solution
1. read from a string input (from stdin)
2. read from a stream (such as /dev/fd/1, stdin too, but you need to know if you are piped). [How do I know the python script is called in a pipe](https://stackoverflow.com/questions/35247817/is-there-a-way-for-a-python-script-to-know-if-it-is-being-piped-to)?

In this mode, no requirement for `git`. When I invoke cz in non-git repo, it will complain

```
$ cz check file-name
fatal: not a git repository (or any of the parent directories): .git
```
## Additional context

current spec

```
usage: cz check [-h]
                [--commit-msg-file COMMIT_MSG_FILE | --rev-range REV_RANGE]

optional arguments:
  -h, --help            show this help message and exit
  --commit-msg-file COMMIT_MSG_FILE
                        ask for the name of the temporal file that contains
                        the commit message. Using it in a git hook script:
                        MSG_FILE=$1
  --rev-range REV_RANGE
                        a range of git rev to check. e.g, master..HEAD
```

Besides the issue I mentioned, I think this spec is not clear about the **positional** argument.

Just let me know if you guys have the bandwidth or not. I can try to help during weekends

**Comment by Lee-W** [comment]
This seems great! Thanks for the suggestion!
So basically, we have the following three things to do.

1. new feature: pipe stream to `cz check`
2. bug fix: check git project only on git related commands
3. doc(?): unclear positional argument (I don't quite get it. Could you please describe a bit more?)

I'm still working on other issues and might not have time to work on it soon. 😞  But I'd be happy to discuss it with you if you need any help when making this change. 😄 We can also enhance our [Contributing Guide](https://commitizen-tools.github.io/commitizen/contributing/) during the process. 🚀

**Comment by guihao-liang** [comment]
> 3\. **doc(?)**: unclear positional argument (I don't quite get it. Could you please describe a bit more?)

Sure! I mean, when you invoke, `cz check args`, what do you expect `args` to be? Files? And what long option is used by default?

**Comment by Lee-W** [comment]
Got it! Thanks! I'll add it after I finish what I'm working on.

**Comment by Lee-W** [comment]
@guihao-liang We add a new argument for `cz check`. It's not exactly the same as the solution you describe, but it might be helpful.
https://github.com/commitizen-tools/commitizen/pull/254

**Comment by Lee-W** [comment]
Thanks to @jordanSu . This feature is now released on 2.4.0 🎉
